controller/appcontroller_test.go

```diff
@@ -2191,3 +2193,40 @@ func TestAlreadyAttemptSync(t *testing.T) {
 		assert.False(t, attempted)
 	})
 }
+
+func assertDurationAround(t *testing.T, expected time.Duration, actual time.Duration) {
+	delta := time.Second / 2
+	assert.GreaterOrEqual(t, expected, actual-delta)
+	assert.LessOrEqual(t, expected, actual+delta)
+}
+
+func TestSelfHealExponentialBackoff(t *testing.T) {
```

Comment:
```suggestion
func TestSelfHealExponentialBackoff(t *testing.T) {
```

This is a great addition. I'm just wondering that we can add some more testcases for this test using the Table Driven approach. WDYT?

Reply:
Agree, added more test cases including nil `app.Status.OperationState.FinishedAt`